File: app/src/main/java/io/githup/limvot/mangagaga/SettingsActivity.scala
package io.githup.limvot.mangagaga;

import android.app.Activity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import org.scaloid.common._

class SettingsActivity extends SActivity {

    implicit val tag = LoggerTag("Scala Setting Activity")
    SettingsManager.loadSettings();
    var historySizeText: SEditText = null

    onCreate{

        contentView = new SScrollView() {

            this += new SVerticalLayout {
        
            val buttonClearHistory = SButton("Clear History",MangaManager.clearHistory())
            val buttonClearCache = SButton("Clear Cache",Utilities.clearCache())
            val buttonClearFavorites = SButton("Clear Favroites",MangaManager.clearFavorites())
            val buttonClearSaved = SButton("Clear Saved Chapters",MangaManager.clearSaved())
            val buttonClearAll = SButton("Clear All", clearAll)
            val buttonCheckUpdate: SButton = SButton("Check for Updates", update)
            val historySize = STextView("Number of entries to save in history")
            historySizeText = SEditText(SettingsManager.getHistorySize.toString)
            historySizeText.setRawInputType(2)

            historySizeText.afterTextChanged {changeHistorySize(historySizeText)}

            val cacheSize = STextView("Number of pages to prefetch")
            val cacheSizeText = SEditText(SettingsManager.getCacheSize.toString)


            cacheSizeText.afterTextChanged { changeCacheSize(cacheSizeText) }
            cacheSizeText.setRawInputType(2)

            }

        }
    }

    def clearAll() {
        Utilities.clearCache()
        MangaManager.clearHistory()
        MangaManager.clearFavorites()
        MangaManager.clearSaved()
    }

    def update() = Utilities.checkForUpdates(this)

    def changeHistorySize(textView : SEditText) {
      try {
        var newSize = Integer.parseInt(textView.toString)
        SettingsManager.setHistorySize(newSize)
      }
      catch{
        case e:NumberFormatException => { Log.d("Settings Changed","Unable to parse history size!")}
      }
    }
    
    def changeCacheSize(textView : SEditText) {
      try {
        var newSize = Integer.parseInt(textView.toString)
        SettingsManager.setCacheSize(newSize)
      }
      catch{
        case e:NumberFormatException => { Log.d("Settings Changed","Unable to parse cache size!")}
      }
    }
}

File: app/src/main/java/io/githup/limvot/mangagaga/SettingsActivity.kt
package io.githup.limvot.mangagaga

import org.jetbrains.anko.*
import org.jetbrains.anko.sdk25.coroutines.onClick

import android.os.Bundle
import android.app.Activity

class SettingsActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        verticalLayout {

            button("Clear History")            { onClick { Boss.clearHistory() } }
            button("Clear Cache")              { onClick { Utilities.clearCache() } }
            button("Clear Favorites")          { onClick { Boss.clearFavorites() } }
            button("Clear Saved Chapters")     { onClick { Boss.clearSaved() } }

            button("Refresh Scripts from Git") { onClick { doAsync { Utilities.gitToScripts()
                                                                     ScriptManager.init() } } }

            button("Clear All")                { onClick { Boss.clearHistory()
                                                           Utilities.clearCache()
                                                           Boss.clearFavorites()
                                                           Boss.clearSaved() } }

            textView("Number of entries to save in history")
            val history_size_entry = editText(SettingsManager.getHistorySize().toString())

            textView("Number of pages to prefetch")
            val cache_size_entry = editText(SettingsManager.getCacheSize().toString())

            textView("http(s) address of git repo to push to")
            val git_url_entry = editText(SettingsManager.getGitURL())

            button("Save") { onClick {
                SettingsManager.setHistorySize(history_size_entry.text.toString().toInt())
                SettingsManager.setCacheSize(cache_size_entry.text.toString().toInt())
                SettingsManager.setGitURL(git_url_entry.text.toString())
            } }
        }
    }
}
